liberty/LibertyReader.cc

```diff
@@ -3959,21 +3959,26 @@ LibertyReader::seqPortNames(LibertyGroup *group,
 			    bool &has_size,
 			    int &size)
 {
-  int i = 0;
   out_name = nullptr;
   out_inv_name = nullptr;
   size = 1;
   has_size = false;
-  for (LibertyAttrValue *value : *group->params()) {
-    if (i == 0)
-      out_name = value->stringValue();
-    else if (i == 1)
-      out_inv_name = value->stringValue();
-    else if (i == 2) {
-      size = static_cast<int>(value->floatValue());
+  if (group->params()->size() == 2) {
+    out_name = group->firstName();
+    out_inv_name = group->secondName();
+  }
+  else if (group->params()->size() == 3) {
+    LibertyAttrValue *third_value = (*group->params())[2];
+    if (third_value->isFloat()) {
+      out_name = group->firstName();
+      out_inv_name = group->secondName();
+      size = static_cast<int>(third_value->floatValue());
       has_size = true;
     }
-    i++;
+    else {
+      out_name = group->secondName();
```

Comment:
comment
// in_port, out_port, out_port_inv, bus_size

Reply:
There is no `bus_size`, that would be 4 params. Should I add a case for that? Haven't seen it in the wild yet, but might be good to have

Reply:
right; I copied too much in the comment.